modules/extensions-support/src/test/java/org/mule/runtime/module/extension/internal/runtime/operation/InterceptingOperationMessageProcessorTestCase.java

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.module.extension.internal.runtime.operation;
+
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.CoreMatchers.sameInstance;
+import static org.junit.Assert.assertThat;
+import static org.junit.Assert.fail;
+import static org.mockito.Answers.RETURNS_DEEP_STUBS;
+import static org.mockito.Matchers.any;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+import org.mule.runtime.core.api.MuleMessage;
+import org.mule.runtime.core.api.processor.MessageProcessor;
+import org.mule.runtime.extension.api.runtime.operation.InterceptingCallback;
+import org.mule.tck.size.SmallTest;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.runners.MockitoJUnitRunner;
+
+@SmallTest
+@RunWith(MockitoJUnitRunner.class)
+public class InterceptingOperationMessageProcessorTestCase extends AbstractOperationMessageProcessorTestCase
+{
+
+    @Mock(answer = RETURNS_DEEP_STUBS)
+    private InterceptingCallback interceptingCallback;
+
+    @Mock
+    private MessageProcessor next;
+
+    @Override
+    @Before
+    public void before() throws Exception
+    {
+        super.before();
+        when(interceptingCallback.shouldProcessNext()).thenReturn(true);
+        when(operationExecutor.execute(any())).thenReturn(interceptingCallback);
+    }
+
+    @Override
+    protected OperationMessageProcessor createOperationMessageProcessor()
+    {
+        InterceptingOperationMessageProcessor messageProcessor = new InterceptingOperationMessageProcessor(extensionModel, operationModel, configurationName,
+                                                                                                           target, resolverSet, extensionManager);
+
+        messageProcessor.setListener(next);
+
+        return messageProcessor;
+    }
+
+    @Test
+    public void nextExecuted() throws Exception
+    {
+        org.mule.runtime.core.api.MuleEvent resultEvent = mock(org.mule.runtime.core.api.MuleEvent.class);
```

Comment:
i'm using both the new and the old type of events because MessageProcessor API hasn't been updated yet. I try not to import the old API so that it's easier to visualize when the MUle API is more mature and these components ready for update